pkg/network/amqp.go

```diff
@@ -58,11 +67,25 @@ func (a *Amqp) Stop() {
 }
 
 // PublishPersistentMessage sends a persistent message to RabbitMQ
-func (a *Amqp) PublishPersistentMessage(exchange, exchangeType, key string, body []byte, headers amqp.Table) error {
-	err := a.declareExchange(exchange, exchangeType)
+func (a *Amqp) PublishPersistentMessage(exchange, exchangeType, key string, msg interface{}, options *MessageOptions) error {
```

Comment:
```suggestion
func (a *Amqp) PublishPersistentMessage(exchange, exchangeType, key string, msg MsgInterface, options *MessageOptions) error {
```

Reply:
Can you add a interface on network/msg.go?
This layer shouldn't know if we are using json or not.